Scripts/RemediationScripts/Remediate-PTAControls.ps1

```diff
@@ -0,0 +1,915 @@
+        {
+            Write-Host "Az.Accounts module is present ... " -ForegroundColor $([Constants]::MessageType.Info)
+        }
+        else
+        {
+             Write-Host "Installing Az.Accounts module with required version(2.5.4)..." -ForegroundColor $([Constants]::MessageType.Info)
+             Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery' -Force -ErrorAction Stop        
+        }    
+    }
+    else
+    {
+        Write-Host "Installing Az.Accounts module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery'  -Force -ErrorAction Stop
+    }
+
+    if($availableModules.Name -contains "AzureAD")
+    {
+        Write-Host "AzureAD module is present ..." -ForegroundColor $([Constants]::MessageType.Info)
+    }
+    else
+    {
+        Write-Host "Installing AzureAD module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name AzureAD -Scope CurrentUser -Repository "PSGallery" -Force -ErrorAction Stop
+    }
+}
+
+
+function Create-PIMForPermanentRoleAssignments
+{
+    <#
+        .SYNOPSIS
+        Remediates 'Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access , Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access_RG' Control.
+
+        .DESCRIPTION
+        Remediates 'Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access , Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access_RG' Control.
+        To create PIM role assignment(s) with SC-ALT account for critical permanent role assignment(s) and PIM non SCALT role assignment(s) remove critical permanent role assignment(s) for which the PIM role assignment(s) is successfully created in the Subscription or in resource group. 
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription to be remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful remediation without any prompts.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+        
+        .PARAMETER DryRun
+        Specifies a dry run of the actual remediation.
+        
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the remediation.
+
+        .INPUTS
+        None. You cannot pipe objects to Create-PIMForPermanentRoleAssignments.
+
+        .OUTPUTS
+        None. Create-PIMForPermanentRoleAssignments does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Create-PIMForPermanentRoleAssignments -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        .EXAMPLE
+        PS> Create-PIMForPermanentRoleAssignments -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -resourcegroup -DryRun
```

Comment:
Please fix `-resourcegroup` as per the conventions followed with others. Also, why is this required?